libsolutil/JSON.h

```diff
@@ -67,4 +67,67 @@ std::string jsonPrint(Json::Value const& _input, JsonFormat const& _format);
 /// \return \c true if the document was successfully parsed, \c false if an error occurred.
 bool jsonParseStrict(std::string const& _input, Json::Value& _json, std::string* _errs = nullptr);
 
+namespace detail
+{
+
+template<typename T>
+struct helper;
+
+#define DEFINE_HELPER(TYPE, CHECK_TYPE, CONVERT_TYPE)                                                     \
+	template<>                                                                                            \
+	struct helper<TYPE>                                                                                   \
+	{                                                                                                     \
+		static bool ofType(Json::Value const& _input, std::string const& _name)                           \
+		{                                                                                                 \
+			return _input[_name].CHECK_TYPE();                                                            \
```

Comment:
In macros the arguments should be parenthesized whenever possible to avoid weird errors when you use an expression.

Not sure if this will compile in C++, but it it will, it would be safer:
```suggestion
			return _input[_name].(CHECK_TYPE)();                                                            \
```

Reply:
True! But sadly this does not compile.

Reply:
ok. Fine then.